src/access.d

```diff
@@ -486,3 +505,36 @@ extern (C++) bool symbolIsVisible(Dsymbol origin, Dsymbol s)
 {
     return symbolIsVisible(origin.getAccessModule(), s);
 }
+
+/**
+ * Same as above but also checks for protected symbols visible from scope `sc`.
+ * Used for qualified name lookup.
+ *
+ * Params:
+ *  sc = lookup scope
+ *  s = symbol to check for visibility
+ * Returns: true if s is visible by origin
+ */
+extern (C++) bool symbolIsVisible(Scope *sc, Dsymbol s)
```

Comment:
If we'd pass the current scope from Scope.search to Dsymbol.search, then it might be possible to replace the above overloads w/ this newly added function, but I'd like to try that separately of this regression fix.